food-truck-api/src/main/java/food/truck/api/endpoint/NotificationEndpoint.java

```diff
@@ -0,0 +1,115 @@
+package food.truck.api.endpoint;
+
+import food.truck.api.notification.Notification;
+import food.truck.api.notification.NotificationService;
+import food.truck.api.notification.NotificationView;
+import food.truck.api.reviews_and_subscriptions.SubscriptionService;
+import food.truck.api.truck.TruckService;
+import food.truck.api.user.User;
+import lombok.NonNull;
+import lombok.Value;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@Log4j2
+@RestController
+public class NotificationEndpoint {
+    @Autowired
+    private NotificationService notificationService;
+
+    @Autowired
+    private SubscriptionService subscriptionService;
+
+    @Autowired
+    TruckService truckService;
+
+    @Value
+    public static class SendNotificationParams {
+        long truckId;
+        @NonNull
+        String message;
+        // TODO: different types of notifications
+    }
+
+    @Secured({"ROLE_USER"})
+    @PostMapping("/truck/notification")
+    public boolean sendNotificationForTruck(
+        @AuthenticationPrincipal User u,
+        @RequestBody SendNotificationParams notificationParams
+    ) {
+        var t = truckService.findTruckById(notificationParams.truckId);
+        if (t.isEmpty()) {
+            return false;
+        }
+        var truck = t.get();
+        if (!truck.getUserId().equals(u.getId()) && u.getIsOwner()) {
+            return false;
+        }
+        var subscriptions = subscriptionService.findSubsByTruck(truck);
+        notificationService.saveNotification(subscriptions, notificationParams.message);
+        return true;
+    }
+
+    @Secured({"ROLE_USER"})
+    @GetMapping("/truck/{truckId}/notifications")
+    public List<NotificationView> sendNotificationForTruck(
```

Comment:
This method looks like it's retrieving notifications, not sending them. Why is it named `sendNotificationForTruck`?

Reply:
You're right. I must have copy-pasted some code and forgotten to rename the function.